### PR TITLE
docs(api): improve openapi documentation for dash, charts and queries

### PR DESCRIPTION
**Pull request by dpgaspar**
### CATEGORY

Choose one

- [ ] Bug Fix
- [X] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [X] Documentation

### SUMMARY
Add descriptions for endpoints and fields to improve API's documentation. Also added an improved check on allowed datasource_type checks.

Chart POST schema:
Screenshot 2020-05-04 at 16 13 17

Dashboard POST schema:
Screenshot 2020-05-04 at 16 14 26

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by craig-rueda — APPROVED** [review]

File: superset/charts/api.py [hunk 1]
```diff
@@ -509,20 +516,14 @@ def thumbnail(
             FileWrapper(screenshot), mimetype="image/png", direct_passthrough=True
         )
 
-    def add_apispec_components(self, api_spec: APISpec) -> None:
-        for chart_type in CHART_DATA_SCHEMAS:
-            api_spec.components.schema(
-                chart_type.__name__, schema=chart_type,
-            )
-        super().add_apispec_components(api_spec)
-
```
Comment (villebro) [inline superset/charts/api.py:518]: Are these now populated automatically?

File: superset/charts/schemas.py [hunk 2]
```diff
@@ -38,35 +95,74 @@ def validate_json(value: Union[bytes, bytearray, str]) -> None:
...
-    datasource_name = fields.String(allow_none=True)
-    dashboards = fields.List(fields.Integer())
+    """
+    Schema to add a new chart.
+    """
+
+    slice_name = fields.String(
+        description=slice_name_description, required=True, validate=Length(1, 250)
+    )
+    description = fields.String(description=description_description, allow_none=True)
```
Comment (villebro) [inline superset/charts/schemas.py:105]: I tried googling when to use `required` and when to use `allow_none`, but wasn't really able to find anything conclusive. Has this come up previously? I just used `required` everywhere, but now I'm not so sure anymore..

**Review by villebro — COMMENTED** [review]
A few quick comments

File: superset/charts/api.py (hunk 1 above)
Comment (dpgaspar) [reply superset/charts/api.py:518]: moved it to base, we can now set these kind of things using `openapi_spec_component_schemas` class attr.

File: superset/charts/schemas.py (hunk 2 above)
Comment (dpgaspar) [reply superset/charts/schemas.py:105]: There are some slight differences, my understanding is that `required`, requires that a field is present, but the field can be present but the value is `null` for example

**Review by villebro — APPROVED** [review]
LGTM 👍

**Comment by dpgaspar** [comment]
Thks @villebro!

**Pull request merged**